extras/check_code_style.sh

```diff
@@ -0,0 +1,215 @@
+#!/bin/bash
+
+# static code analysis
+START_SHA=master
+END_SHA=HEAD
+
+NEST_SRC=.
+CPPCHECK=cppcheck
+VERA=vera++
+VERA_HOME=/usr/local/lib/vera++
+
+CLANG_FORMAT=clang-format
+INCREMENTAL=false
+
+ONLY_FILE=
+
+#
+# usage [exit_code bad_option]
+#
+usage ()
+{
+    if test $1 -ne 0 ; then
+        echo "Unknown option: $2"
+    fi
+
+    cat <<EOF
+Usage: check_code_style.sh [options ...]"
+
+Setup of Tooling is explained here: 
+    https://nest.github.io/nest-simulator/coding_guidelines_c++
+
+Options:
+
+    --help               Print program options and exit
+    --incremental        Do analysis one file after another.
+    --file=/path/to/file Perform the static analysis on this file only.
```

Comment:
Doc should specify that this must be a relative path and that the script must either be run from the nest root directory, or `--nest-src` must also be given. See details on problems that may arise otherwise below.